src/usr/local/containerbase/tools/v2/rye.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+function install_tool () {
+  local versioned_tool_path
+  local arch
+  local name
+  local version
+  local platform
+  local repo
+
+  arch=${ARCHITECTURE}
+  name=$TOOL_NAME
+  version=$TOOL_VERSION
+  platform=$(uname -s)
+  repo="astral-sh/$name"
+
+  if [[ $platform == "Linux" ]]; then
+    platform="linux"
+  fi
+
+  if [[ $arch == armv8* ]] || [[ $arch == arm64* ]] || [[ $arch == aarch64* ]]; then
+    arch="aarch64"
+  elif [[ $arch == i686* ]]; then
+    arch="x86"
+  fi
+
+  binary="${name}-${arch}-${platform}"
+  file_name="${binary}.gz"
+  base_url="https://github.com/${repo}/releases/download/${version}"
+  file_url="${base_url}/${file_name}"
+
+  checksum_file=$(get_from_url "${file_url}.sha256")
+  # get checksum from file
+  expected_checksum=$(cut -d' ' -f1  "${checksum_file}")
+
+  file=$(get_from_url "${file_url}" "${file_name}" "${expected_checksum}" "sha256sum")
+
+  versioned_tool_path=$(create_versioned_tool_path)
+  create_folder "${versioned_tool_path}/bin"
+
+  gunzip -c "${file}" > "${TEMP_DIR}/rye"
+  chmod +x "${TEMP_DIR}/rye"
+
+  export RYE_HOME="${versioned_tool_path}"
+  "${TEMP_DIR}/rye" self install --yes
```

Comment:
is this downloading additional files? what the effects?

Reply:
@viceice From what I can gather this is at the final step in the "official installer". This "installer" is a script which you are supposed to pipe directly into bash/zsh/etc. I don't think that's a great way to install things, so I'm happy @vhdirk implemented it without that mechanism. You can find the curl into shell command on the official Rye site here: https://rye.astral.sh

I have also copied the current version of the script below for archival purposes should anyone stumble upon this PR in the future :) **Note the `self install` command on line 80**

```bash
    1 #!/usr/bin/env bash
    2 set -euo pipefail
    3 
    4 # Wrap everything in a function so that a truncated script
    5 # does not have the chance to cause issues.
    6 __wrap__() {
    7 
    8 # allow overriding the version
    9 VERSION=${RYE_VERSION:-latest}
   10 # allow overriding the install option
   11 INSTALL_OPTION=${RYE_INSTALL_OPTION:-""}
   12 
   13 REPO=astral-sh/rye
   14 PLATFORM=`uname -s`
   15 ARCH=`uname -m`
   16 
   17 if [[ $PLATFORM == "Darwin" ]]; then
   18   PLATFORM="macos"
   19 elif [[ $PLATFORM == "Linux" ]]; then
   20   PLATFORM="linux"
   21 fi
   22 
   23 if [[ $ARCH == armv8* ]] || [[ $ARCH == arm64* ]] || [[ $ARCH == aarch64* ]]; then
   24   ARCH="aarch64"
   25 elif [[ $ARCH == i686* ]]; then
   26   ARCH="x86"
   27 fi
   28 
   29 BINARY="rye-${ARCH}-${PLATFORM}"
   30 
   31 # Oddly enough GitHub has different URLs for latest vs specific version
   32 if [[ $VERSION == "latest" ]]; then
   33   DOWNLOAD_URL=https://github.com/${REPO}/releases/latest/download/${BINARY}.gz
   34 else
   35   DOWNLOAD_URL=https://github.com/${REPO}/releases/download/${VERSION}/${BINARY}.gz
   36 fi
   37 
   38 echo "This script will automatically download and install rye (${VERSION}) for you."
   39 if [ "x$(id -u)" == "x0" ]; then
   40   echo "warning: this script is running as root.  This is dangerous and unnecessary!"
   41 fi
   42 
   43 if ! hash curl 2> /dev/null; then
   44   echo "error: you do not have 'curl' installed which is required for this script."
   45   exit 1
   46 fi
   47 
   48 if ! hash gunzip 2> /dev/null; then
   49   echo "error: you do not have 'gunzip' installed which is required for this script."
   50   exit 1
   51 fi
   52 
   53 TEMP_FILE=`mktemp "${TMPDIR:-/tmp}/.ryeinstall.XXXXXXXX"`
   54 TEMP_FILE_GZ="${TEMP_FILE}.gz"
   55 
   56 cleanup() {
   57   rm -f "$TEMP_FILE"
   58   rm -f "$TEMP_FILE_GZ"
   59 }
   60 
   61 trap cleanup EXIT
   62 HTTP_CODE=$(curl -SL --progress-bar "$DOWNLOAD_URL" --output "$TEMP_FILE_GZ" --write-out "%{http_code}")
   63 if [[ ${HTTP_CODE} -lt 200 || ${HTTP_CODE} -gt 299 ]]; then
   64   echo "error: platform ${PLATFORM} (${ARCH}) is unsupported."
   65   exit 1
   66 fi
   67 
   68 rm -f "$TEMP_FILE"
   69 gunzip "$TEMP_FILE_GZ" 
   70 chmod +x "$TEMP_FILE"
   71 
   72 # Detect when the file cannot be executed due to NOEXEC /tmp.  Taken from rustup
   73 # https://github.com/rust-lang/rustup/blob/87fa15d13e3778733d5d66058e5de4309c27317b/rustup-init.sh#L158-L159
   74 if [ ! -x "$TEMP_FILE" ]; then
   75   printf '%s\n' "Cannot execute $TEMP_FILE (likely because of mounting /tmp as noexec)." 1>&2
   76   printf '%s\n' "Please copy the file to a location where you can execute binaries and run it manually." 1>&2
   77   exit 1
   78 fi
   79 
   80 "$TEMP_FILE" self install $INSTALL_OPTION
   81 
   82 }; __wrap__
```